### ISSUE TITLE
Booting a already provisioned VM doesn't mount Chef vboxsf folders in Vagrant 1.7.2

### ISSUE THREAD

**Issue by andytson** [mitchellh/vagrant#5140]
On the 2nd vagrant up of a VM with chef-solo configuration for path configuration, the vboxsf folders that are usually mounted don't appear. This issue occurs in Vagrant 1.7.2 but not 1.7.1.

Steps to reproduce:

    vagrant remote add opscode-centos-6.6 http://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_centos-6.6_chef-provisionerless.box
    vagrant init opscode-centos-6.6
    mkdir roles
    mkdir cookbooks

Add to the bottom of Vagrantfile:

      config.omnibus.chef_version = :latest
      config.vm.provision :chef_solo do |chef|
        chef.cookbooks_path = "cookbooks"
        chef.roles_path = "roles"
      end
.

    vagrant up
    vagrant ssh -c 'mount'
    vagrant reload
    vagrant ssh -c 'mount'

The 2nd mount is missing the cookbooks and roles mounts from the first mount call

**Comment by sethvargo** [comment]
@mitchellh is looks like we aren't caching all the synced folders?

**Comment by databus23** [comment]
I'm having similar issues. I can't get vagrant 1.7.2 to mount the chef related folders at all after a `vagrant reload`.

**Comment by andytson** [comment]
Is it somehow related to this that was fixed? https://github.com/mitchellh/vagrant/issues/4988

**Comment by andytson** [comment]
A guess at the issue is it's possibly related to this caching: true here, causing what @sethvargo mentions as the chef folders not being added to the cache:
https://github.com/mitchellh/vagrant/pull/5121/files#diff-4b59695a716db3067f2a3b5d98bff333R37

**Comment by khobbits** [comment]
I'm having some major issues with this, on 1.7.2.

I have a few chef recipes I want to run on every 'up', not just on provision and these now bomb.

"config.vm.provision :chef_zero, run: "always" do |chef|"

**Comment by fletchowns** [comment]
I think this is the error message I'm seeing as a results of this bug. The initial ```vagrant up``` works fine but after a ```vagrant halt```, ```vagrant up```, ```vagrant provision``` I see the following:

> ==> default: [2015-01-13T21:20:32+00:00] ERROR: The expanded run list includes nonexistent roles: app_server
> ==> default: [2015-01-13T21:20:32+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

**Comment by valchonedelchev** [comment]
The tests also confirmed that there is an issue like @andytson said. I am on commit: 21736aa5d4abeb550b52e61eedd230b82cb4cf47

```
bash$ rake acceptance:run COMPONENTS='provider/virtualbox/synced_folder'

Failures:

  1) provider/virtualbox/synced_folder it should behave like provider/synced_folder properly configures synced folder types
     Failure/Error: expect(result.exit_code).to eql(0)
       
       expected: 0
            got: 1
       
       (compared using eql?)
```

**Comment by bitwisekaizen** [comment]
I just hit this too.  Will revert back to 1.7.1 for now.

**Comment by valchonedelchev** [comment]
Same here.

**Comment by ozzyogkush** [comment]
I'm seeing this same issue after upgrading to 1.7.2 - roles, cookbooks, environments, etc are all empty inside the VM filesystem.

**Comment by etki** [comment]
@ozzyogkush afaik deleting `.vagrant/default/synced_folders` using vagrant-triggers plugin before every boot provides a temporary solution. I'm not sure if i tested it against 1.7.2, though.

**Comment by andytson** [comment]
What remains to be done about this ticket. It seems to be a critical issue that has no fix for almost 6 months. I see there has been a fix supplied for specific provisioners.

**Comment by triccardi-systran** [comment]
Same issue here, thanks @etki for the workaround!

How to work around this bug:

```bash
vagrant plugin install vagrant-triggers
```
In `Vagrantfile`:
```ruby
  unless Vagrant.has_plugin?("vagrant-triggers")
    raise "vagrant-triggers plugin missing, cannot automatically work around vagrant 1.7.2 bug #5140."
  end
  config.trigger.before :up do
    info "Workaround vagant 1.7.2 issue #5140"
    run "rm -f .vagrant/machines/default/virtualbox/synced_folders"
  end
```

**Comment by fletchowns** [comment]
Is this a duplicate of #5199 ?

**Comment by etki** [comment]
@fletchowns given it only a diagonal look, but seems so

**Comment by dkinzer** [comment]
@fletchowns @etki  It looks like the same issue, this issue was posted 8 days prior to this one, so maybe #5199 is the dupe.

**Comment by mitchellh** [comment]
Fixed in #5199
